packages/components/system/src/system.types.tsx

```diff
@@ -55,11 +55,15 @@ export type MergeWithAs<
     as?: AsComponent
   }
 
-export type ComponentWithAs<Component extends As, Props extends object = {}> = {
+export type ComponentWithAs<
+  Component extends As,
+  Props extends object = {},
+  Omitted extends string = never,
+> = {
   <AsComponent extends As = Component>(
     props: MergeWithAs<
-      React.ComponentProps<Component>,
-      React.ComponentProps<AsComponent>,
+      Omit<React.ComponentProps<Component>, Omitted>,
+      Omit<React.ComponentProps<AsComponent>, Omitted>,
```

Comment:
from the performance PoV using `Omit` twice here is not that great - but with the current shape of this type it has to be used on both of type arguments